model/skipgraph/identifier.go

```diff
@@ -4,15 +4,40 @@ import (
 	"bytes"
 	"encoding/hex"
 	"fmt"
+	"github.com/go-playground/validator/v10"
 )
 
 const IdentifierSizeBytes = 32
 
-type ComparisonResult string
+var validate *validator.Validate
 
-const CompareEqual ComparisonResult = "compare-equal"
-const CompareGreater ComparisonResult = "compare-greater"
-const CompareLess ComparisonResult = "compare-less"
+func init() {
+	validate = validator.New()
+}
+
+const (
+	CompareEqual   = "compare-equal"
+	CompareGreater = "compare-greater"
+	CompareLess    = "compare-less"
+)
+
+type ComparisonResult struct {
+	// made this field unexported to ensure that only the constructor func is used
+	// to create an instance of this type
+	result string `validate:"oneof=compare-equal compare-greater compare-less"`
+}
+
+func (cr ComparisonResult) Result() string {
+	return cr.result
+}
+func NewComparisonResult(s string) (*ComparisonResult, error) {
+	cr := ComparisonResult{s}
+	err := validate.Struct(cr)
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to validate the comparison result upon instantiation: %w", err)
```
Let's add context when bubbling up an error.

Reply:
Of course, done https://github.com/thep2p/skipgraph-go/pull/18/commits/789dbd55471e216a657254c5ca2e69390f1cf938 !